cobra/flux_analysis/sampling.py

```diff
@@ -113,33 +150,29 @@ def generate_fva_warmup(self, solver=None, **solver_args):
         **solver_args
             Additional arguments passed to the solver.
         """
-
-        solver = solver_dict[get_solver_name() if solver is None else solver]
-        lp = solver.create_problem(self.model)
-        for i, r in enumerate(self.model.reactions):
-            solver.change_variable_objective(lp, i, 0.0)
-
         self.n_warmup = 0
-        self.warmup = np.zeros((2 * len(self.model.reactions),
-                                len(self.model.reactions)))
-
-        for sense in ("minimize", "maximize"):
-            for i, r in enumerate(self.model.reactions):
-                # Omit fixed reactions
-                if self.fixed[i]:
-                    pass
-                solver.change_variable_objective(lp, i, 1.0)
-                solver.solve_problem(lp, objective_sense=sense, **solver_args)
-                sol = solver.format_solution(lp, self.model).x
-                if not sol:
-                    pass
-                # some solvers do not enforce bounds too much -> we reconstrain
-                sol = np.maximum(sol, self.bounds[0, ])
-                sol = np.minimum(sol, self.bounds[1, ])
-                self.warmup[self.n_warmup, ] = sol
-                self.n_warmup += 1
-                # revert objective
-                solver.change_variable_objective(lp, i, 0.)
+        idx = np.hstack([self.fwd_idx, self.rev_idx])
+        self.warmup = np.zeros((len(idx), len(self.model.variables)))
+        self.model.objective = S.Zero
+        self.model.objective.direction = "max"
+        variables = self.model.variables
+        for i in idx:
+            # Omit fixed reactions
+            if self.fixed[i]:
+                pass
+            self.model.objective.set_linear_coefficients({variables[i]: 1})
+            self.model.solver.optimize()
+            if self.model.solver.status != "optimal":
+                pass
+            primals = self.model.solver.primal_values
```

Comment:
this throws AssertionError for me for iMM904 and yeast7, would help to catch those (although I guess need to throw away the whole sample then :/ )..